### PR TITLE
UF-9932: Add municipalityId

### PR DESCRIPTION
**Pull request by cheezi0747**

### CODE REVIEW

File: src/integration-test/java/apptest/IntegrationTest.java [hunk 1]
```diff
@@ -101,57 +107,65 @@ void test01_standardPostBatchJob() throws JsonProcessingException, ClassNotFound
...
 		setupCall()
 			.withHttpMethod(GET)
-			.withServicePath("/archived/attachments?batchHistoryId=" + postBatchHistory.getId())
-			.withExpectedResponseStatus(NOT_FOUND)
+			.withServicePath(ARCHIVED_PATH + "?batchHistoryId=" + postBatchHistory.getId())
+			.withExpectedResponseStatus(OK)
 			.withExpectedResponse("response.json")
-			.sendRequestAndVerifyResponse();
+			.sendRequestAndVerifyResponse()
+			.verifyAllStubs();
```
Comment (josc04) [inline src/integration-test/java/apptest/IntegrationTest.java:136]: I'm not sure, but isn't the stubs verified in the sendRequestAndVerifyResponse-method
```
	public AbstractAppTest sendRequestAndVerifyResponse() {
		return sendRequest().verifyStubs();
	}
```
?

File: src/main/java/se/sundsvall/byggrarchiver/service/ArchiveHistoryService.java [hunk 2]
```diff
@@ -227,4 +233,10 @@ private boolean isArchived(final ArchiveHistory archiveHistory) {
 		return (archiveHistory != null) && (archiveHistory.getArchiveId() != null);
 	}
 
+	public List<ArchiveHistoryResponse> getArchiveHistories(final ArchiveStatus archiveStatus, final Long batchHistoryId, final @ValidMunicipalityId String municipalityId) {
```
Comment (josc04) [inline src/main/java/se/sundsvall/byggrarchiver/service/ArchiveHistoryService.java:236]: The `@ValidMunicipalityId`-annotation is not necessary here, right?

Comment (cheezi0747) [reply src/main/java/se/sundsvall/byggrarchiver/service/ArchiveHistoryService.java:236]: nope. Intellij autogeneration of methods believe so though. Deleting.

File: src/test/java/se/sundsvall/byggrarchiver/api/model/ArchiveHistoryResponseTest.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+		final var documentName = "documentName";
+		final var documentType = "documentType";
+		final var archiveId = "archiveId";
+		final var archiveUrl = "archiveUrl";
+		final var archiveStatus = ArchiveStatus.COMPLETED;
+		final var timestamp = LocalDateTime.now();
+		final var batchHistory = new BatchHistoryResponse();
+
+		// Act
+		final var batchHistoryResponse = ArchiveHistoryResponse.builder()
```
Comment (josc04) [inline src/test/java/se/sundsvall/byggrarchiver/api/model/ArchiveHistoryResponseTest.java:56]: OCD comment, but rename variable to archiveHistoryResponse 😄

File: src/test/java/se/sundsvall/byggrarchiver/service/scheduler/ArchiveSchedulerTest.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+		when(mockByggrArchiverService.runBatch(originalStart, end, batchTrigger, municipalityId))
+			.thenReturn(BatchHistoryResponse.builder().withId(randomLong()).build());
+
+		when(mockSchedulerProperties.municipalityIds()).thenReturn(List.of(municipalityId));
+
+		//Act
+		archiverScheduler.archive();
+
+		// Assert
+		verify(mockByggrArchiverService, times(1)).runBatch(originalStart, end, batchTrigger, municipalityId);
```
Comment (josc04) [inline src/test/java/se/sundsvall/byggrarchiver/service/scheduler/ArchiveSchedulerTest.java:52]: `times(1)` is redundant

**Review by josc04 — APPROVED** [review]

**Pull request merged**